### ISSUE TITLE
How to batch update submissions titles?

### ISSUE THREAD

**Issue by romainpoirier** [verbb/formie#1085]
### Question

I have a form which I forgot to update the `Submission Title Format` before it goes live. So submissions have been recorded with the default `Timestamp` title.

I have now updated the `Submission Title Format` to something more relevant, like `First Name Last Name <Email>`, but previous submitted entries titles aren't changed.

As there's no batch resave (like `./craft resave/entries`), I have resave them manually, but unfortunately the title is still not updated.

Is there any way to do so? Thank you!

**Comment by engram-design** [comment]
You can use the same resave command `./craft resave/formie-submissions` (typing `./craft` will show you all available console commands).

Refer to https://github.com/verbb/formie/issues/1066 and https://github.com/verbb/formie/issues/1070

**Comment by romainpoirier** [comment]
Thank you. I just tested it, but this doesn't work. The title are not updated. However, I don't get any error.
I also tried to run `./craft resave/formie-forms` before just in case, but without any effect.

**Comment by engram-design** [comment]
Title's aren't forced when you save a submission, the `Submission Title Format` is purely used for the default submission. This allows you override the title to something you prefer, or need as a special case (refer to my [comment](https://github.com/verbb/formie/issues/1066#issuecomment-1234922430)).

As such, because your submissions already have a title, they won't be changed based on a later-changed `Submission Title Format`.

Try: `resave/formie-submissions --set=title --to="={firstName} {lastName} {email}"` - if your fields have these same handles

**Comment by romainpoirier** [comment]
Thank you, I tried this:
`./craft resave/formie-submissions --set=title --to="={firstName} {lastName} {email}"`

But even if these handles are correct, and titles are not empty, I get this error on each resaved line:
```
Resaving 10 elements ...
    - [1/10] Resaving John Doe < john@doe.com > (7386) ... failed: Title cannot be blank.
    - [2/10] Resaving 2022-09-05 16:40:10 (6549) ... failed: Title cannot be blank.
    - [3/10] Resaving 2022-09-05 16:26:07 (6545) ... failed: Title cannot be blank.
    - [4/10] Resaving 2022-09-05 15:44:28 (6541) ... failed: Title cannot be blank.
    - [5/10] Resaving 2022-09-05 15:43:27 (6537) ... failed: Title cannot be blank.
    - [6/10] Resaving 2022-09-05 15:43:12 (6533) ... failed: Title cannot be blank.
    - [7/10] Resaving 2022-09-05 15:42:50 (6529) ... failed: Title cannot be blank.
    - [8/10] Resaving 2022-09-05 15:41:20 (6525) ... failed: Title cannot be blank.
    - [9/10] Resaving 2022-09-05 15:40:50 (6521) ... failed: Title cannot be blank.
    - [10/10] Resaving 2022-09-05 14:57:32 (6517) ... failed: Title cannot be blank.
```

I also ran this after, just in case, but without any change:
`./craft resave/formie-submissions`

**Comment by engram-design** [comment]
Do you have fields with handles `firstName`, `lastName`, and `email`? If those fields don't exist (or there's no data for those fields in the submission) then it will indeed be a blank string.

**Comment by romainpoirier** [comment]
Yes of course, and they are required fields. However, they are within a `Group` field which handle is `contact`.

So I tried this, but without any change:
`./craft resave/formie-submissions --set=title --to="={contact.firstName} {contact.lastName} {contact.email}"`

**Comment by engram-design** [comment]
Groups act a little differently. `./craft resave/formie-submissions --set=title --to="={contact.one().firstName} {contact.one().lastName} {contact.one().email}"`

**Comment by romainpoirier** [comment]
Thank you, that works!

One more (a bit unrelated) thing:

- I was trying to get title as `First Name Last Name <Email>`, so with `<` and `>` around `Email`.
- When I edit the `Submission Title Format`, I can add  `<` and `>` around `Email` without any space
- On save, and after reload, a space is added automatically around the `Email`, so `< Email >`, I can't delete it

**Comment by engram-design** [comment]
I did initially see this as well, but after some updates with Craft, I can't seem to replicate it.

Just to double check, this is just when creating a submission normally, not using the console commands to update the title? Or both?

**Comment by romainpoirier** [comment]
Yes this is just when creating a submission normally.

**Comment by engram-design** [comment]
Slightly unrelated, as this was just the Vue-based field in the form settings that was causing this, adding extra spaces. Nothing to do with the console command. Fixed for the next release. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.10",
  "...": "..."
}
```

Then run `composer update`. You'll just want to re-save your form settings with the properly formatted "Submission Title Format".

**Comment by engram-design** [comment]
Fixed in [2.0.11](https://github.com/verbb/formie/releases/tag/2.0.11)
